### PR TITLE
feat: add version bump check

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This PR adds a workflow step that checks if the version was bumped so we don't merge something that will not be uploaded to pypi. It also adds a dependency that's used in the tests into the tests dependency array.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the version of the package in pyproject.toml

### CODE REVIEW

File: .github/workflows/test-package.yaml [hunk 1]
```diff
@@ -9,6 +9,17 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v4
+      - name: check version
+        id: vbump
+        uses: rgryta/Check-Bump@main
+        with:
+          method: 'toml'
+          prefix: 'v'
```
Comment (ghislainbourgeois) [inline .github/workflows/test-package.yaml:17]: Is this prefix enough? The field is named "version" in the pyproject.toml file.

Comment (kayra1) [reply .github/workflows/test-package.yaml:17]: yea, if you look at the last 2 workflow runs it can tell that version was bumped after the last commit

Comment (kayra1) [reply .github/workflows/test-package.yaml:17]: also that prefix is for the output string for the action, `vbump.outputs.version`, it's not used when analyzing the pyproject.toml file.

Comment (ghislainbourgeois) [reply .github/workflows/test-package.yaml:17]: Ok, thank you!

**Pull request merged**